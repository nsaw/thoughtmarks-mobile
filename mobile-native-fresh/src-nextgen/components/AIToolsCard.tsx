import React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';

interface AIToolsCardProps {
  slotType?: string;
  tool?: {
    id: string;
    name: string;
    description: string;
    status: 'available' | 'busy' | 'offline';
  };
}

export const AIToolsCard: React.FC<AIToolsCardProps> = ({ 
  slotType = 'DEFAULT',
  tool 
}) => {
  const getStatusColor = (status: string) => {
    switch (status) {
      case 'available': return '#4CAF50';
      case 'busy': return '#FF9800';
      case 'offline': return '#F44336';
      default: return '#666';
    }
  };

  return (
    <TouchableOpacity 
      style={{
        backgroundColor: '#333',
        padding: 16,
        borderRadius: 8,
        marginVertical: 4,
        borderLeftWidth: 4,
        borderLeftColor: getStatusColor(tool?.status || 'offline'),
      }}
    >
      <View>
        <Text style={{ color: 'white', fontSize: 16, fontWeight: '600' }}>
          {tool?.name || 'AI Tool'}
        </Text>
        <Text style={{ color: '#ccc', fontSize: 14, marginTop: 4 }}>
          {tool?.description || 'AI tool description...'}
        </Text>
        <Text style={{ color: '#666', fontSize: 12, marginTop: 8 }}>
          Slot: {slotType} | Status: {tool?.status || 'offline'}
        </Text>
      </View>
    </TouchableOpacity>
  );
}; 