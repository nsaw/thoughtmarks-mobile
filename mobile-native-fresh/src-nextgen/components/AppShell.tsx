// AppShell.tsx - JSX baseline role prop hydration
import React from 'react';
import { View } from 'react-native';
import type { RoleProps } from '/Users/sawyer/gitSync/tm-mobile-cursor/mobile-native-fresh/src-reference-complete/types/roles';

interface AppShellProps extends RoleProps {
  children: React.ReactNode;
}

export const AppShell: React.FC<AppShellProps> = ({ layoutRole, contentRole, interactiveRole, children }) => {
  return (
    <View testID="AppShell-Root">
      {children}
    </View>
  );
}; 