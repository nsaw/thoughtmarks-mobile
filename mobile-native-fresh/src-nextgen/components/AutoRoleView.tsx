import React from 'react';
import { View, TouchableOpacity, AccessibilityRole, Role } from 'react-native';

import { usePerformanceMonitor } from '../utils/PerformanceMonitor';

interface AutoRoleViewProps {
  children: React.ReactNode;
  layoutRole?: string;
  style?: any;
  onPress?: () => void;
  accessibilityRole?: AccessibilityRole;
  accessible?: boolean;
  accessibilityLabel?: string;
  testID?: string;
  role?: Role;
  ariaLabel?: string;
  ariaDescribedBy?: string;
  ariaHidden?: boolean;
  ariaExpanded?: boolean;
  ariaPressed?: boolean;
  ariaChecked?: boolean;
  ariaSelected?: boolean;
  ariaDisabled?: boolean;
  ariaRequired?: boolean;
  ariaInvalid?: boolean;
  ariaLive?: 'off' | 'polite' | 'assertive';
  ariaAtomic?: boolean;
  ariaRelevant?: string;
  ariaBusy?: boolean;
  ariaControls?: string;
  ariaOwns?: string;
  ariaLabelledBy?: string;
  // Performance monitoring props
  enablePerformanceMonitoring?: boolean;
  componentName?: string;
  // Error boundary props
  errorBoundary?: boolean;
  fallback?: React.ComponentType<{ error?: Error }>;
  // Validation props
  validateProps?: boolean;
  strictMode?: boolean;
}

export const AutoRoleView: React.FC<AutoRoleViewProps> = ({
  children,
  layoutRole: _layoutRole, // eslint-disable-line no-unused-vars
  style,
  onPress,
  accessibilityRole,
  accessible,
  accessibilityLabel,
  testID,
  role,
  ariaLabel,
  ariaDescribedBy,
  ariaHidden,
  ariaExpanded,
  ariaPressed,
  ariaChecked,
  ariaSelected,
  ariaDisabled,
  ariaRequired,
  ariaInvalid,
  ariaLive,
  ariaAtomic,
  ariaRelevant,
  ariaBusy,
  ariaControls,
  ariaOwns,
  ariaLabelledBy,
  enablePerformanceMonitoring = false,
  componentName = 'AutoRoleView',
  errorBoundary = false,
  fallback,
  validateProps = false,
  strictMode = false,
  ...props
}) => {
  // Prop validation in strict mode
  if (validateProps || strictMode) {
    if (!children) {
      console.warn('AutoRoleView: children prop is required');
    }
    if (accessible && !accessibilityLabel) {
      console.warn('AutoRoleView: accessibilityLabel is required when accessible is true');
    }
  }

  // Performance monitoring
  const { recordComponentRender } = usePerformanceMonitor();
  React.useEffect(() => {
    if (enablePerformanceMonitoring) {
      const startTime = Date.now();
      const cleanup = recordComponentRender(componentName, 'nextgen')(Date.now() - startTime);
      return cleanup;
    }
  }, [enablePerformanceMonitoring, componentName, recordComponentRender]);

  // Error boundary wrapper
  if (errorBoundary && fallback) {
    return React.createElement(fallback, { error: undefined });
  }

  const accessibilityProps = {
    accessible,
    accessibilityRole,
    accessibilityLabel,
    testID,
    role,
    ariaLabel,
    ariaDescribedBy,
    ariaHidden,
    ariaExpanded,
    ariaPressed,
    ariaChecked,
    ariaSelected,
    ariaDisabled,
    ariaRequired,
    ariaInvalid,
    ariaLive,
    ariaAtomic,
    ariaRelevant,
    ariaBusy,
    ariaControls,
    ariaOwns,
    ariaLabelledBy,
  };

  if (onPress) {
    return (
      <TouchableOpacity
        {...accessibilityProps}
        style={style}
        onPress={onPress}
        {...props}
      >
        {children}
      </TouchableOpacity>
    );
  }

  return (
    <View
      {...accessibilityProps}
      style={style}
      {...props}
    >
      {children}
    </View>
  );
};
