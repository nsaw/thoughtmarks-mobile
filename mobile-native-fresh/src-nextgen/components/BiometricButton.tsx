import React from 'react';
import { TouchableOpacity, ActivityIndicator, StyleSheet, AccessibilityRole } from 'react-native';
import { ThemeColors } from '../types/theme';
import { Text as CustomText } from './Text';

export interface BiometricButtonProps {
  icon: string;
  label: string;
  onPress: () => void;
  loading?: boolean;
  disabled?: boolean;
  style?: object;
  accessibilityLabel?: string;
  accessibilityHint?: string;
  accessibilityRole?: AccessibilityRole;
}

export const BiometricButton: React.FC<BiometricButtonProps> = ({
  icon,
  label,
  onPress,
  loading = false,
  disabled = false,
  style,
  accessibilityLabel,
  accessibilityHint,
  accessibilityRole = 'button' as AccessibilityRole,
}) => {
  const colors: ThemeColors = {
    background: '#FFFFFF',
    surface: '#F8F9FA',
    text: '#1A1A1A',
    textSecondary: '#6C757D',
    primary: '#007AFF',
    error: '#DC3545',
    border: '#DEE2E6',
  };

  return (
    <TouchableOpacity
      style={[
        styles.button,
        {
          backgroundColor: disabled ? colors.textSecondary : colors.surface,
          borderColor: colors.border,
        },
        style,
      ]}
      onPress={onPress}
      disabled={disabled || loading}
      accessibilityLabel={accessibilityLabel}
      accessibilityHint={accessibilityHint}
      accessibilityRole={accessibilityRole}
    >
      {loading ? (
        <ActivityIndicator color={colors.primary} />
      ) : (
        <>
          <CustomText variant="body" style={styles.icon}>
            {icon}
          </CustomText>
          <CustomText variant="body" style={{ ...styles.label, color: colors.text }}>
            {label}
          </CustomText>
        </>
      )}
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
    gap: 8,
    borderWidth: 1,
    borderRadius: 8,
    paddingHorizontal: 16,
  },
  icon: {
    fontSize: 20,
  },
  label: {
    fontSize: 16,
    fontWeight: '500',
  },
}); 
