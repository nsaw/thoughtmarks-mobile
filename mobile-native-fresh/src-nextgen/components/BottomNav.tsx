import React from 'react';
import { View, Text, TouchableOpacity, StyleSheet } from 'react-native';
import { useTheme } from '../theme/ThemeProvider';

export interface BottomNavItem {
  id: string;
  label: string;
  icon: string;
  route: string;
}

export interface BottomNavProps {
  items: BottomNavItem[];
  activeRoute: string;
  onRouteChange: (route: string) => void;
  style?: any;
}

export const BottomNav: React.FC<BottomNavProps> = ({
  items,
  activeRoute,
  onRouteChange,
  style,
}) => {
  const { theme } = useTheme();

  const styles = StyleSheet.create({
    container: {
      backgroundColor: theme.colors.surface,
      borderTopWidth: 1,
      borderTopColor: theme.colors.border,
      flexDirection: 'row',
      paddingHorizontal: theme.spacing.sm,
      paddingVertical: theme.spacing.xs,
    },
    item: {
      flex: 1,
      alignItems: 'center',
      paddingVertical: theme.spacing.sm,
      borderRadius: 8,
    },
    activeItem: {
      backgroundColor: theme.colors.accent,
    },
    icon: {
      fontSize: 20,
      marginBottom: theme.spacing.xs,
    },
    activeIcon: {
      color: theme.colors.onAccent,
    },
    inactiveIcon: {
      color: theme.colors.textSecondary,
    },
    label: {
      fontSize: theme.typography.fontSize.caption,
      fontWeight: theme.typography.fontWeight.medium as any,
    },
    activeLabel: {
      color: theme.colors.onAccent,
    },
    inactiveLabel: {
      color: theme.colors.textSecondary,
    },
  });

  return (
    <View style={[styles.container, style]}>
      {items.map((item) => {
        const isActive = activeRoute === item.route;
        return (
          <TouchableOpacity
            key={item.id}
            style={[styles.item, isActive && styles.activeItem]}
            onPress={() => onRouteChange(item.route)}
            activeOpacity={0.7}
          >
            <Text style={[
              styles.icon,
              isActive ? styles.activeIcon : styles.inactiveIcon
            ]}>
              {item.icon}
            </Text>
            <Text style={[
              styles.label,
              isActive ? styles.activeLabel : styles.inactiveLabel
            ]}>
              {item.label}
            </Text>
          </TouchableOpacity>
        );
      })}
    </View>
  );
}; 