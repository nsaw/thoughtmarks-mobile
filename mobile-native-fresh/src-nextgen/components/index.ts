// src-nextgen/components/index.ts
// Nextgen components with role assignments

export { Text } from './Text';
export { Button } from './Button';
export { TagChip } from './TagChip';
export { ModernHeader } from './ModernHeader';
export { BottomNav } from './BottomNav';
export { ThoughtmarkCard } from './ThoughtmarkCard';
export { TaskCard } from './TaskCard';
export { AIToolsCard } from './AIToolsCard'; 