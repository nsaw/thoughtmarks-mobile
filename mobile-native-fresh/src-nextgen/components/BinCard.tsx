import React, { useMemo } from 'react';
import { View, Text, StyleSheet, TouchableOpacity } from 'react-native';
import { AutoRoleView } from './AutoRoleView';

interface Bin {
  id: string;
  name: string;
  content: string[];
  createdAt: Date;
}

interface BinCardProps {
  bin: Bin;
  onPress?: () => void;
}

// Memoization â€” Cache expensive computations in components and hooks
export const BinCard: React.FC<BinCardProps> = ({ bin, onPress }) => {
  // MEMOIZATION: Cache expensive selector calculations
  const summary = useMemo(() => summarizeBinContent(bin), [bin.id]);
  
  const itemCount = useMemo(() => bin.content.length, [bin.content.length]);
  
  const formattedDate = useMemo(() => {
    return bin.createdAt.toLocaleDateString();
  }, [bin.createdAt]);

  return (
    <TouchableOpacity onPress={onPress}>
      <AutoRoleView role="content" style={styles.container}>
        <Text style={styles.title}>{bin.name}</Text>
        <Text style={styles.summary}>{summary}</Text>
        <View style={styles.metadata}>
          <Text style={styles.count}>{itemCount} items</Text>
          <Text style={styles.date}>{formattedDate}</Text>
        </View>
      </AutoRoleView>
    </TouchableOpacity>
  );
};

// Expensive computation function that benefits from memoization
const summarizeBinContent = (bin: Bin): string => {
  // Simulate expensive computation
  const words = bin.content.join(' ').split(' ');
  const uniqueWords = new Set(words);
  const averageLength = words.reduce((sum, word) => sum + word.length, 0) / words.length;
  
  return `${uniqueWords.size} unique words, avg length: ${averageLength.toFixed(1)}`;
};

const styles = StyleSheet.create({
  container: {
    padding: 16,
    backgroundColor: '#fff',
    borderRadius: 8,
    marginVertical: 8,
    marginHorizontal: 16,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 4,
    elevation: 3,
  },
  title: {
    fontSize: 18,
    fontWeight: 'bold',
    marginBottom: 8,
  },
  summary: {
    fontSize: 14,
    color: '#666',
    marginBottom: 12,
  },
  metadata: {
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  count: {
    fontSize: 12,
    color: '#999',
  },
  date: {
    fontSize: 12,
    color: '#999',
  },
});

export default BinCard; 