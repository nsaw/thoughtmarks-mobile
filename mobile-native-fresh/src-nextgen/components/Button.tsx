import React from 'react';
import { Text } from 'react-native';

import { useTheme } from '../theme/ThemeProvider';
import { ActionButton } from '../components/AutoRoleView';

interface ButtonProps {
  title?: string;
  children?: React.ReactNode;
  onPress?: () => void;
  style?: any;
  disabled?: boolean;
  variant?: 'primary' | 'secondary' | 'outline' | 'ghost' | 'destructive' | 'brand';
  size?: 'small' | 'medium' | 'large' | 'sm' | 'md' | 'lg' | 'icon';
  leftIcon?: React.ReactNode;
  interactiveRole?: 'button-action' | 'button-secondary' | 'button-destructive';
}

export function Button({ 
  title,
  children,
  style,
  variant = 'primary',
  leftIcon,
  interactiveRole = 'button-action',
  ...props 
}: ButtonProps) {
  const theme = useTheme();

  // Use title if provided, otherwise use children
  const buttonContent = title || children;

  // Apply variant and size styles through the style prop
  const buttonStyle = {
    ...style,
    // Add variant-specific styles here if needed
    ...(variant === 'outline' && { borderWidth: 1, borderColor: theme.designTokens.colors.border }),
    ...(variant === 'ghost' && { backgroundColor: 'transparent' }),
    ...(variant === 'destructive' && { backgroundColor: theme.designTokens.colors.accent }),
    ...(variant === 'brand' && { backgroundColor: theme.designTokens.colors.accent }),
  };

  // Determine role based on variant and interactiveRole prop
  const role = variant === 'destructive' ? 'button-destructive' : interactiveRole;

  return (
    <ActionButton 
      style={buttonStyle} 
      {...props}
    >
      {leftIcon}
      <Text style={{ 
        color: theme.designTokens.colors.buttonText,
        fontSize: theme.designTokens.typography.fontSize.body,
        fontWeight: theme.designTokens.typography.fontWeight.medium as any,
      }}>
        {buttonContent}
      </Text>
    </ActionButton>
  );
} 