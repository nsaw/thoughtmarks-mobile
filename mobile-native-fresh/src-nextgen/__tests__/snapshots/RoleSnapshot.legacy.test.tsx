import React from 'react';
import renderer from 'react-test-renderer';
import { Card, Tag } from '../../components/system';

describe('JSX Snapshot Baseline â€” Role Components', () => {
  it('renders Card correctly', () => {
    const tree = renderer.create(<Card>Card baseline</Card>).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders Tag correctly', () => {
    const tree = renderer.create(<Tag>Tag baseline</Tag>).toJSON();
    expect(tree).toMatchSnapshot();
  });
}); 