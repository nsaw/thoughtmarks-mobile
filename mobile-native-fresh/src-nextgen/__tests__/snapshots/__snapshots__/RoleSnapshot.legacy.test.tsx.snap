// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Role component snapshot validation Snapshot baseline validation - file structure 1`] = `
{
  "Card": {
    "exists": true,
    "modified": "2025-07-24T03:27:11.828Z",
    "path": "/Users/sawyer/gitSync/tm-mobile-cursor/mobile-native-fresh/src/components/ui/Card.tsx",
    "size": 3611,
  },
  "TagChip": {
    "exists": true,
    "modified": "2025-07-18T00:16:32.386Z",
    "path": "/Users/sawyer/gitSync/tm-mobile-cursor/mobile-native-fresh/src/components/ui/TagChip.tsx",
    "size": 3015,
  },
}
`;
